internal/plans/planfile/wrapped.go

```diff
@@ -0,0 +1,45 @@
+package planfile
+
+import "github.com/hashicorp/terraform/internal/cloud/cloudplan"
+
+// WrappedPlanFile is a sum type that represents a saved plan, loaded from a
+// file path passed on the command line. If the specified file was a thick local
+// plan file, the Local field will be populated; if it was a bookmark for a
+// remote cloud plan, the Cloud field will be populated. In both cases, the
+// other field is expected to be nil. Finally, the outer struct is also expected
+// to be used as a pointer, so that a nil value can represent the absence of any
+// plan file.
+type WrappedPlanFile struct {
+	Local *Reader
+	Cloud *cloudplan.SavedPlanBookmark
```

Comment:
I might consider making both of these fields unexported and then changing the two accessor methods to be like `func (w *WrappedPlanFile) Local() (*Reader, bool)` (and similar for `Cloud`) so that the caller can't so easily avoid checking which variant is selected, and thus it's hopefully more obvious how to use this API correctly. I don't feel super strongly about it, but just suggesting it in case it seems viable and worthwhile to you.

Reply:
Ah, thanks! I'll wait to hear from the other reviewers, but it seems like probably a reasonable improvement to make.

Reply:
OK, I did that refactor! 👍🏼 And as for setting both fields — I made a pair of constructor functions for tests that need to fake up one of these structs, and those plus OpenWrapped are the only things allowed to actually build one of these; that should make it impossible to get ahold of a malformed one.